### PR TITLE
Fix Android and iOS builds

### PR DESCRIPTION
**Pull request by TomTasche**
Thank you so much for maintaining this project! It helped us a lot for an app we're developing: https://opendocument.app/

There's one small change I had to apply to make it work on iOS and one bigger change that was necessary to make it build on Android too. I'd be happy to see those changes go upstream and help others too.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1025,7 +1032,7 @@ if (BUILD_STATIC)
   add_library(cryptopp-static STATIC ${cryptopp_LIBRARY_SOURCES})
   cryptopp_target_compile_properties(cryptopp-static)
   if (NOT ${CMAKE_VERSION} VERSION_LESS "2.8.11")
-    target_include_directories(cryptopp-static PUBLIC $<BUILD_INTERFACE:${SRC_DIR}> $<INSTALL_INTERFACE:include>)
+    target_include_directories(cryptopp-static PUBLIC $<BUILD_INTERFACE:${SRC_DIR}/..> $<INSTALL_INTERFACE:include>)
```
Comment (TomTasche) [inline CMakeLists.txt]: That was necessary to make it build for iOS. Unfortunately I can't remember the exact scenario, but it's been building fine for us for a few months now.

**Comment by noloader** [comment]
Thanks @TomTasche..

One question... Where does ``ANDROID_ABI` come from? Is that a CMake variable?

One comment... I believe that should be `${ANDROID_NDK_ROOT}`. Also see [ Recommended NDK Directory?](https://groups.google.com/forum/#!msg/android-ndk/qZjhOaynHXc/2ux2ZZdxy2MJ) on the Android NDK mailing list.

**Comment by TomTasche** [comment]
Found out about ANDROID_ABI on [StackOverflow](https://stackoverflow.com/a/53422788/198996) and decided to use it as an indicator if the current build is for Android. It seems to be coming from the Android CMake toolchain.

I'll give ANDROID_NDK_ROOT a try and report back.

**Comment by noloader** [comment]
It looks like you can use `CMAKE_SYSTEM_NAME` to detect Android builds.

Also see [`cmake-variables(7)`](https://cmake.org/cmake/help/latest/variable/ANDROID.html) in the Cmake docs.

**Comment by TomTasche** [comment]
ANDROID_NDK_ROOT doesn't work unfortunately (empty on my machine). I could use CMAKE_ANDROID_NDK though: https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling-for-android

**Comment by noloader** [comment]
Thanks @TomTasche,

I'd like to setup the situation where things just work folks doing thing the right way (i.e., setting `ANDROID_NDK_ROOT` and friends).

Is it possible for you to wire-in `ANDROID_NDK_ROOT`? Something like:

```
if (NOT ANDROID_NDK)
    if (ANDROID_NDK_ROOT)
        set (ANDROID_NDK "${ANDROID_NDK_ROOT}")
    endif()
endif()
```

Let me send you a collaborator invite so you can modify things without waiting for me. It also ensures there are admins available if I get hit by a bus.

**Comment by TomTasche** [comment]
Sorry for the radio silence on my end! I saw there was another pull request addressing Android, I'm glad that was merged. I tested it and it works fine for me, therefore I'm going to close this pull request for now. I might have to reopen it in the future for the iOS change that's included here too, but I'm still investigating that...

**Pull request closed without merging**